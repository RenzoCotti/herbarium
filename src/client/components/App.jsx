import React, { Component } from "react";
import { Route, HashRouter } from "react-router-dom";

import Navbar from "./Navbar";
import PlantDetail from "./PlantDetail";
import Browse from "./Browse";
import Footbar from "./Footbar";
import Home from "./Home";

class App extends Component {
  // async getRandom() {
  //   this.setState({ loading: true });
  //   let res = await fetch("/random");
  //   let jplant = await res.json();

  //   this.setState({ loading: false });
  //   return jplant;
  // }

  render() {
    return (
      <HashRouter>
        <div>
          <Navbar />

          <div className="content">
            <Route exact path="/" component={Home} />
            <Route path="/browse" component={Browse} />
            <Route path="/random" component={PlantDetail} />
            <Route path="/identify" component={PlantDetail} />
          </div>

          <Footbar />
        </div>
      </HashRouter>
    );
  }
}

export default App;
