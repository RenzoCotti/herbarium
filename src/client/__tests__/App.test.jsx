import App from "../components/App";
import React from "react";
import { shallow } from "../enzyme";

describe("App", () => {
  it("renders correctly", () => {
    const nav = shallow(<App />);
    expect(nav).toMatchSnapshot();
  });
  it("should have 3 children", () => {
    const nav = shallow(<App />);
    expect(nav.children()).toHaveLength(3);
  });
});
