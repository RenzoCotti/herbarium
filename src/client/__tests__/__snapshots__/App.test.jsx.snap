// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Navbar />,
        <div
          className="main-container"
        >
          <Media
            images={
              Object {
                "flowers": "https://cdn.shopify.com/s/files/1/1766/2959/products/Peppermint_4_400x400.jpg?v=1535245781",
                "leaves": "https://images-na.ssl-images-amazon.com/images/I/51TeuM4HqJL.jpg",
                "overall": "https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/Mentha-piperita.JPG/330px-Mentha-piperita.JPG",
              }
            }
            name="Peppermint"
          />
          <Information
            plant={
              Object {
                "blooming": Array [
                  "mid summer",
                  "late summer",
                ],
                "common": "Peppermint",
                "flowers": Array [
                  "purple",
                ],
                "habitat": "gravel",
                "harvest": "spring",
                "height": 0.6,
                "images": Object {
                  "flowers": "https://cdn.shopify.com/s/files/1/1766/2959/products/Peppermint_4_400x400.jpg?v=1535245781",
                  "leaves": "https://images-na.ssl-images-amazon.com/images/I/51TeuM4HqJL.jpg",
                  "overall": "https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/Mentha-piperita.JPG/330px-Mentha-piperita.JPG",
                },
                "latin": "Mentha Piperita",
                "leaves": Object {
                  "height": 2.5,
                  "margin": "toothed",
                  "properties": Array [
                    "dark green",
                    "red",
                  ],
                  "shape": "lanceolate",
                  "width": 6,
                },
                "region": "Tundra",
                "shape": "bush",
                "stem": "smooth",
                "type": "herbaceous plant",
              }
            }
          />
        </div>,
        <Footbar />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Media
              images={
                Object {
                  "flowers": "https://cdn.shopify.com/s/files/1/1766/2959/products/Peppermint_4_400x400.jpg?v=1535245781",
                  "leaves": "https://images-na.ssl-images-amazon.com/images/I/51TeuM4HqJL.jpg",
                  "overall": "https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/Mentha-piperita.JPG/330px-Mentha-piperita.JPG",
                }
              }
              name="Peppermint"
            />,
            <Information
              plant={
                Object {
                  "blooming": Array [
                    "mid summer",
                    "late summer",
                  ],
                  "common": "Peppermint",
                  "flowers": Array [
                    "purple",
                  ],
                  "habitat": "gravel",
                  "harvest": "spring",
                  "height": 0.6,
                  "images": Object {
                    "flowers": "https://cdn.shopify.com/s/files/1/1766/2959/products/Peppermint_4_400x400.jpg?v=1535245781",
                    "leaves": "https://images-na.ssl-images-amazon.com/images/I/51TeuM4HqJL.jpg",
                    "overall": "https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/Mentha-piperita.JPG/330px-Mentha-piperita.JPG",
                  },
                  "latin": "Mentha Piperita",
                  "leaves": Object {
                    "height": 2.5,
                    "margin": "toothed",
                    "properties": Array [
                      "dark green",
                      "red",
                    ],
                    "shape": "lanceolate",
                    "width": 6,
                  },
                  "region": "Tundra",
                  "shape": "bush",
                  "stem": "smooth",
                  "type": "herbaceous plant",
                }
              }
            />,
          ],
          "className": "main-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "images": Object {
                "flowers": "https://cdn.shopify.com/s/files/1/1766/2959/products/Peppermint_4_400x400.jpg?v=1535245781",
                "leaves": "https://images-na.ssl-images-amazon.com/images/I/51TeuM4HqJL.jpg",
                "overall": "https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/Mentha-piperita.JPG/330px-Mentha-piperita.JPG",
              },
              "name": "Peppermint",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "plant": Object {
                "blooming": Array [
                  "mid summer",
                  "late summer",
                ],
                "common": "Peppermint",
                "flowers": Array [
                  "purple",
                ],
                "habitat": "gravel",
                "harvest": "spring",
                "height": 0.6,
                "images": Object {
                  "flowers": "https://cdn.shopify.com/s/files/1/1766/2959/products/Peppermint_4_400x400.jpg?v=1535245781",
                  "leaves": "https://images-na.ssl-images-amazon.com/images/I/51TeuM4HqJL.jpg",
                  "overall": "https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/Mentha-piperita.JPG/330px-Mentha-piperita.JPG",
                },
                "latin": "Mentha Piperita",
                "leaves": Object {
                  "height": 2.5,
                  "margin": "toothed",
                  "properties": Array [
                    "dark green",
                    "red",
                  ],
                  "shape": "lanceolate",
                  "width": 6,
                },
                "region": "Tundra",
                "shape": "bush",
                "stem": "smooth",
                "type": "herbaceous plant",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Navbar />,
          <div
            className="main-container"
          >
            <Media
              images={
                Object {
                  "flowers": "https://cdn.shopify.com/s/files/1/1766/2959/products/Peppermint_4_400x400.jpg?v=1535245781",
                  "leaves": "https://images-na.ssl-images-amazon.com/images/I/51TeuM4HqJL.jpg",
                  "overall": "https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/Mentha-piperita.JPG/330px-Mentha-piperita.JPG",
                }
              }
              name="Peppermint"
            />
            <Information
              plant={
                Object {
                  "blooming": Array [
                    "mid summer",
                    "late summer",
                  ],
                  "common": "Peppermint",
                  "flowers": Array [
                    "purple",
                  ],
                  "habitat": "gravel",
                  "harvest": "spring",
                  "height": 0.6,
                  "images": Object {
                    "flowers": "https://cdn.shopify.com/s/files/1/1766/2959/products/Peppermint_4_400x400.jpg?v=1535245781",
                    "leaves": "https://images-na.ssl-images-amazon.com/images/I/51TeuM4HqJL.jpg",
                    "overall": "https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/Mentha-piperita.JPG/330px-Mentha-piperita.JPG",
                  },
                  "latin": "Mentha Piperita",
                  "leaves": Object {
                    "height": 2.5,
                    "margin": "toothed",
                    "properties": Array [
                      "dark green",
                      "red",
                    ],
                    "shape": "lanceolate",
                    "width": 6,
                  },
                  "region": "Tundra",
                  "shape": "bush",
                  "stem": "smooth",
                  "type": "herbaceous plant",
                }
              }
            />
          </div>,
          <Footbar />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Media
                images={
                  Object {
                    "flowers": "https://cdn.shopify.com/s/files/1/1766/2959/products/Peppermint_4_400x400.jpg?v=1535245781",
                    "leaves": "https://images-na.ssl-images-amazon.com/images/I/51TeuM4HqJL.jpg",
                    "overall": "https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/Mentha-piperita.JPG/330px-Mentha-piperita.JPG",
                  }
                }
                name="Peppermint"
              />,
              <Information
                plant={
                  Object {
                    "blooming": Array [
                      "mid summer",
                      "late summer",
                    ],
                    "common": "Peppermint",
                    "flowers": Array [
                      "purple",
                    ],
                    "habitat": "gravel",
                    "harvest": "spring",
                    "height": 0.6,
                    "images": Object {
                      "flowers": "https://cdn.shopify.com/s/files/1/1766/2959/products/Peppermint_4_400x400.jpg?v=1535245781",
                      "leaves": "https://images-na.ssl-images-amazon.com/images/I/51TeuM4HqJL.jpg",
                      "overall": "https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/Mentha-piperita.JPG/330px-Mentha-piperita.JPG",
                    },
                    "latin": "Mentha Piperita",
                    "leaves": Object {
                      "height": 2.5,
                      "margin": "toothed",
                      "properties": Array [
                        "dark green",
                        "red",
                      ],
                      "shape": "lanceolate",
                      "width": 6,
                    },
                    "region": "Tundra",
                    "shape": "bush",
                    "stem": "smooth",
                    "type": "herbaceous plant",
                  }
                }
              />,
            ],
            "className": "main-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "images": Object {
                  "flowers": "https://cdn.shopify.com/s/files/1/1766/2959/products/Peppermint_4_400x400.jpg?v=1535245781",
                  "leaves": "https://images-na.ssl-images-amazon.com/images/I/51TeuM4HqJL.jpg",
                  "overall": "https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/Mentha-piperita.JPG/330px-Mentha-piperita.JPG",
                },
                "name": "Peppermint",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "plant": Object {
                  "blooming": Array [
                    "mid summer",
                    "late summer",
                  ],
                  "common": "Peppermint",
                  "flowers": Array [
                    "purple",
                  ],
                  "habitat": "gravel",
                  "harvest": "spring",
                  "height": 0.6,
                  "images": Object {
                    "flowers": "https://cdn.shopify.com/s/files/1/1766/2959/products/Peppermint_4_400x400.jpg?v=1535245781",
                    "leaves": "https://images-na.ssl-images-amazon.com/images/I/51TeuM4HqJL.jpg",
                    "overall": "https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/Mentha-piperita.JPG/330px-Mentha-piperita.JPG",
                  },
                  "latin": "Mentha Piperita",
                  "leaves": Object {
                    "height": 2.5,
                    "margin": "toothed",
                    "properties": Array [
                      "dark green",
                      "red",
                    ],
                    "shape": "lanceolate",
                    "width": 6,
                  },
                  "region": "Tundra",
                  "shape": "bush",
                  "stem": "smooth",
                  "type": "herbaceous plant",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
